pex/version.py

```diff
@@ -3,9 +3,10 @@
 
 __version__ = '1.4.5'
 
-# NB: If we upgrade to setuptools>=34 pex's bootstrap code in `PEXBuilder` will need an update to
-# include the `packaging` package in the `.bootstrap/` code since we use
-# `packaging.specifiers.SpecifierSet` - indirectly - through `pkg_resources.Requirement.specifier`.
-SETUPTOOLS_REQUIREMENT = 'setuptools>=20.3,<34.0'
+# Versions 34.0.0 through 35.0.2 (last pre-36.0.0) de-vendored dependencies which causes problems
+# for pex code so we exclude that range.
+__exclusions = '!=34.*,!=35.*'
+SETUPTOOLS_REQUIREMENT = 'setuptools>=20.3,<41,{exclusions}'.format(exclusions=__exclusions)
+del __exclusions
```

Comment:
the use and explicit `del` of `__exclusions` struck me as odd here. any reason to not keep the exclusions around as a constant (like `SETUPTOOLS_EXCLUSIONS`) - or just elide and hardcode in the string literal?

Reply:
Agreed - I'll inline. This was half remembering wickmanisms and half wishing for assignment blocks:
```
x = block:
  foo
  # NB: bar is very very worth exposition
  bar

  'baz'

assert x == 'baz'
```